x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```diff
@@ -0,0 +1,459 @@
+        HistogramFieldMapper gpfmMergeWith = (HistogramFieldMapper) mergeWith;
+        if (gpfmMergeWith.ignoreMalformed.explicit()) {
+            this.ignoreMalformed = gpfmMergeWith.ignoreMalformed;
+        }
+    }
+
+    @Override
+    protected String contentType() {
+        return CONTENT_TYPE;
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context, List<IndexableField> fields) throws IOException {
+        throw new UnsupportedOperationException("Parsing is implemented in parse(), this method should NEVER be called");
+    }
+
+    public static class HDRPercentilesFieldType extends MappedFieldType {
+        public HDRPercentilesFieldType() {
+        }
+
+        HDRPercentilesFieldType(HDRPercentilesFieldType ref) {
+            super(ref);
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public MappedFieldType clone() {
+            return new HDRPercentilesFieldType(this);
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
+            failIfNoDocValues();
+            return new IndexFieldData.Builder() {
+
+                @Override
+                public IndexFieldData<?> build(IndexSettings indexSettings, MappedFieldType fieldType, IndexFieldDataCache cache,
+                                               CircuitBreakerService breakerService, MapperService mapperService) {
+
+                    return new IndexHistogramFieldData(indexSettings.getIndex(), fieldType.name()) {
+
+                        @Override
+                        public AtomicHistogramFieldData load(LeafReaderContext context) {
+                            return new AtomicHistogramFieldData() {
+                                @Override
+                                public HistogramValues getHistogramValues() {
+                                    try {
+                                        final BinaryDocValues values = DocValues.getBinary(context.reader(), fieldName);
+                                        return new HistogramValues() {
+                                            @Override
+                                            public boolean advanceExact(int doc) throws IOException {
+                                                return values.advanceExact(doc);
+                                            }
+
+                                            @Override
+                                            public HistogramValue histogram() {
+                                                try {
+                                                    return getHistogramValue(values.binaryValue());
+                                                } catch (IOException e) {
+                                                    throw new IllegalStateException("Cannot load doc value", e);
+                                                }
```

Comment:
Let's make `histogram()` throw an `IOException` for consistency with `BinaryDocValues#binaryValue`?